### PR TITLE
Docker config

### PR DESCRIPTION
**Pull request by adamgraves-choices**

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
Can you fix the codacy issue and rebase if needed. Once done I'll do the review.

**Comment by ajthomascouk** [comment]
@neilbmclaughlin, @st3v3nhunt mentioned we may stop using codacy, whats your thoughts?

**Comment by adamgraves-choices** [comment]
@neilbmclaughlin @beseven This branch is ready to review now.

File: scripts/docker_build_all_tagged_commits.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+TAGS=$(git tag)
+DOCKERFILE='Dockerfile'
+DOCKER_REPO='nhsuk/connecting-to-services'
+PUSH_TO_DOCKER='true'
+
+for TAG in $TAGS; do
+  echo "Building tag: $TAG"
+  git checkout $TAG #2&>1 > /dev/null
+  if [ -e $DOCKERFILE ]; then
+    docker build -t ${DOCKER_REPO}:${TAG} -f ${DOCKERFILE} .
```
Comment (neilbmclaughlin) [inline scripts/docker_build_all_tagged_commits.sh]: SANITISED_TAG=$( echo $TAG | sed 's/\+/_/g' )

File: scripts/docker_build_all_tagged_commits.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+TAGS=$(git tag)
+DOCKERFILE='Dockerfile'
+DOCKER_REPO='nhsuk/connecting-to-services'
+PUSH_TO_DOCKER='true'
```
Comment (neilbmclaughlin) [inline scripts/docker_build_all_tagged_commits.sh]: Parameterise PUSH_TO_DOCKER?

File: scripts/docker_ci.sh [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+PUSH_TO_DOCKER=true
```
Comment (neilbmclaughlin) [inline scripts/docker_ci.sh]: Parameterise PUSH_TO_DOCKER?

File: scripts/docker_ci.sh [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    fold_end "Tagging '$TAG' and pushing to docker hub"
+  done
+
+  fold_end "Tagging and pushing images to docker hub"
+
+fi
+
+info "All builds successful!"
+
+exit 0
```
Comment (neilbmclaughlin) [inline scripts/docker_ci.sh]: Don't think this is needed.

File: scripts/docker_ci.sh [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  SANITISED_BRANCH=$( echo $TRAVIS_BRANCH | sed 's/\//_/g' )
+  TAGS="$TAGS $SANITISED_BRANCH"
+
+  # IF MASTER BRANCH ALWAYS SET THE LATEST TAG
+  if [ "$TRAVIS_BRANCH" = "master" ]; then
+    TAGS="$TAGS latest"
+  fi
+
+  # ADD TAG BRANCH
+  if [ -n "TRAVIS_TAG" ]; then
```
Comment (neilbmclaughlin) [inline scripts/docker_ci.sh]: Should be $TRAVIS_TAG

File: .travis.yml [hunk 6]
```diff
@@ -7,6 +13,7 @@ script:
 - npm run check-coverage
 - npm run upload-coverage-coveralls
 - npm run upload-coverage-codacy
-notifications:
-  slack:
-    secure: cKmjGeAX1JWRw+YRUsAn+IxUTenokF35bP1Uq1revB2AJ1e9oE9fyPaLLdTHwa6tRsqhraC3OM9VRkGkkK2xp8lDifLe3VPsKx0C0EFaFaa3Y5XGM4dtL7YBaIQV45QpXwsO/kMa6wKOyL1wlN9Eo8+9rHpu9mulwRmgUYVmICWvXxD8v1oleXwbNeUfXatVa1x6qt8anF2BEAPvjJ3oSVd87EIBHByYIzkMWgcB0FsWVp19NMAvDsa/DjTs0qb0gi5W4Q/TOjtse5Ycv+esiUo2OsUww2QDaGoATjI39W+1H3IGFSsZWeuSGqZmZnAGbUSfz5fsrbTPEHYpiRGBiZ1YktkuXMWzuMRgD5UklZzPY5LxW7a8BwMo1pERSpT7mWEw7JSenL96d+bPONWjQtvx8imS0VWLhMA9t13AtzPHWRC4VUc0fJyAxUWr+T+8tBMnHocSOXXfdbFrfPfYCDmRE8gr3939+yd9xEGYN26q/4l4aLAufZBQJcFhrSaHVsilTmamIXF/HwWwkmaMQUZtKDIPr6OncEe6vrhwhObceLsLNLmDxobNCyEVBVayaFzOMgDWiYIv1x0nd2pK42L6ZQfXJwe1Ex+kmVm2m39HKcC4vwN92+WDn1noL2G6H0spKXrnHRMcI0Rx2C2F8AJEAuICuyK1YV6eNFAoRig=
+- docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
+- chmod +x ./scripts/*
+- ./scripts/docker_ci.sh
```
Comment (neilbmclaughlin) [inline .travis.yml]: Shouldn't need to chmod here - would be better the chmod the script files and commit. Git preserves the x flag but not permissions.

File: .travis.yml [hunk 7]
```diff
@@ -7,6 +13,7 @@ script:
 - npm run check-coverage
 - npm run upload-coverage-coveralls
 - npm run upload-coverage-codacy
-notifications:
```
Comment (adamgraves-choices) [inline .travis.yml]: Nope, must have been a merge issue... will revert.

File: scripts/docker_build_all_tagged_commits.sh (hunk 2 above)
Comment (adamgraves-choices) [inline scripts/docker_build_all_tagged_commits.sh]: I thought in the first instance we should keep the script simple, and these variables won't change in the first instance. We can easily add 'getopts' later on: https://rsalveti.wordpress.com/2007/04/03/bash-parsing-arguments-with-getopts/

**Pull request merged**